### PR TITLE
Schema, extension metadata and README

### PR DESCRIPTION
**Pull request by ColinMaudry**
This PR implements the following fields:
 
- awards/valueCalculationMethod
- contracts/periodRationale
- parties/details/buyerProfile
- contracts/publicPassengerTransportServicesKilometers
- parties/details/url
- tender/reviewDetails
- tender/valueCalculationMethod

These fields are grouped in this extension as they did not group well with other fields.

### CODE REVIEW

**Comment by ColinMaudry** [comment]
When I started working on this extension, I had an uncommited procurementMethodName codelist in the folder. Should it be somewhere else?

```
Code,Title,Description
open,Open,All interested suppliers can submit a tender
restricted,Restricted,Any economic operator may submit a request to participate in response to a tender but the operators selected by the procuring entity may bid
negotiatedWithCallForCompetition,Negotiated with prior publication of a call for competition,TODO
negotiatedWithoutCallForCompetition,Negotiated without prior publication of a call for competition,TODO
competitiveDialogue,Competitive dialogue,TODO
innovationPartnership,Innovation partnership,TODO
otherSingleStage,Other single stage procedure,TODO
otherMultipleStage,Other multiple stage procedure,TODO
```

**Comment by ColinMaudry** [comment]
**conctract.periodRationale**: Do we use this field specifically for the rationale for the modification of the contract duration (TED F20 semantics are even tighter, it's specifically for FAs) or do we generalize it to the rationale of the contract duration?

If we use the former option, we may rename the field.

**Comment by ColinMaudry** [comment]
**award.valueCalculationMethod**: do we keep the same field name as `tender.valueCalculationMethod,` while

- the semantics of `tender.valueCalculationMethod` are "Method used for calculating the estimated value of the concession." (OK)
- those of `award.valueCalculationMethod` are "Any details relevant to the value of the concession according to article 8(3) of the directive of 2014 on concessions" (not specifically about value calculation)

Shall we rename it `concessionValueDetails`?

**Comment by ColinMaudry** [comment]
**contract.publicPassengerTransportServicesKilometers**: the wording of the description in the schema may be improved. Some regulatory material here: https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A32007R1370#d1e732-1-1 (Article 5, paragraph 4.)

**Comment by ColinMaudry** [comment]
README: given this extension shelters fields that don't group with others, I didn't really know how to contextualize it legally (# Legal context). This section may be improved.

**Comment by jpmckinney** [comment]
> When I started working on this extension, I had an uncommited procurementMethodName codelist in the folder. Should it be somewhere else?

I don't think the current mapping needs a `procurementMethodName` codelist anymore.

> conctract.periodRationale: Do we use this field specifically for the rationale for the modification of the contract duration (TED F20 semantics are even tighter, it's specifically for FAs) or do we generalize it to the rationale of the contract duration?

Let's generalize it, as the F20 restriction isn't semantically relevant.

> award.valueCalculationMethod are "Any details relevant to the value of the concession according to article 8(3) of the directive of 2014 on concessions" (not specifically about value calculation)
> Shall we rename it concessionValueDetails?

It's still about calculating/estimating the value of the concession. [Article 8(3)](https://eur-lex.europa.eu/eli/dir/2014/23/oj#d1e1727-1-1) starts with: "The estimated value of the concession shall be calculated using an objective method specified in the concession documents. When calculating the estimated value of the concession, contracting authorities and contracting entities shall, where applicable, take into account in particular: …"

**Comment by ColinMaudry** [comment]
Thanks! I'll improve a bit the wording and the JSON sample, then I'll request your review.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+sudo: false
```
Comment (jpmckinney) [inline .travis.yml]: Can you remote .travis.yml and copy .github/workflows/lint.yml from https://raw.githubusercontent.com/open-contracting/standard-maintenance-scripts/master/fixtures/lint.yml?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,48 @@
-# ocds_eu_extension
-Extension that implements schemas and codes that are specific to the European law.
+# European Union
+
+Implements schemas and codes that are specific to the European law.
```
Comment (jpmckinney) [inline README.md]: schemas -> fields, remove "the"

Add a paragraph:

> For complete guidance on meeting the disclosure requirements of European law, see [OCDS for European Union](https://standard.open-contracting.org/profiles/eu/master/en/).

File: README.md [hunk 3]
```diff
@@ -1,2 +1,48 @@
...
+
+## Legal context
+
+In the European Union, this extension's fields correspond to [eForms BT-10 and BT-610](https://github.com/eForms/eForms). See [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/) for the correspondences to Tenders Electronic Daily (TED).
```
Comment (jpmckinney) [inline README.md]: Is this all the relevant BTs?

File: extension.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": {
+    "en": "European Union"
+  },
+  "description": {
+    "en": "Implements schemas and codes that are specific to the European law."
```
Comment (jpmckinney) [inline extension.json]: Copy change from readme.

File: release-schema.json [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        }
+      }
+    },
+    "Organization": {
+      "properties": {
+        "details": {
+          "properties": {
+            "url": {
+              "title": "Website URL",
+              "description": "The Web address of the website of the organization.",
```
Comment (jpmckinney) [inline release-schema.json]: Use either "The website of the organization." or "The Web address of the organization."

File: release-schema.json [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            }
+          }
+        }
+      }
+    },
+    "Award": {
+      "properties": {
+        "valueCalculationMethod": {
+          "title": "Value calculation method",
+          "description": "Any details relevant to the value of the concession according to article 8(3) of the directive 2014/23/EU on concession contracts.",
```
Comment (jpmckinney) [inline release-schema.json]: "of Directive 2014/23/EU"

File: release-schema.json [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+          "title": "Contract period rationale",
+          "description": "The rationale for the duration of the contract.",
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "publicPassengerTransportServicesKilometers": {
+          "title": "Kilometers of public passenger transport services",
+          "description": "The size of the public transport network procured in the contract, in kilometers.",
```
Comment (jpmckinney) [inline release-schema.json]: A network isn't being procured, but services on a network. I suggest:

> The number of kilometers of public passenger transport services provided by a public service contract.

Based on:

> Competent authorities may decide not to publish this information where a public service contract concerns an annual provision of less than 50 000 kilometres of public passenger transport services.

From https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A32007R1370

File: release-schema.json [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      "properties": {
+        "periodRationale": {
+          "title": "Contract period rationale",
+          "description": "The rationale for the duration of the contract.",
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "publicPassengerTransportServicesKilometers": {
```
Comment (jpmckinney) [inline release-schema.json:69]: We can add a minimum of 0.

**Comment by jpmckinney** [comment]
We can merge this PR once comments are resolved, but we should also add the codelist modifications to this extension: https://github.com/open-contracting-extensions/european-union/issues/40

File: README.md (hunk 3 above)
Comment (ColinMaudry) [reply README.md]: Probably not, I'll add the others.

File: release-schema.json (hunk 5 above)
Comment (ColinMaudry) [reply release-schema.json]: :spider_web:

File: release-schema.json (hunk 8 above)
Comment (ColinMaudry) [reply release-schema.json:69]: Done

File: release-schema.json (hunk 7 above)
Comment (ColinMaudry) [reply release-schema.json]: Done

File: release-schema.json (hunk 6 above)
Comment (ColinMaudry) [reply release-schema.json]: Done

File: release-schema.json (hunk 5 above)
Comment (ColinMaudry) [reply release-schema.json]: Done

File: extension.json (hunk 4 above)
Comment (ColinMaudry) [reply extension.json]: Done

File: README.md (hunk 2 above)
Comment (ColinMaudry) [reply README.md]: Done

File: .travis.yml (hunk 1 above)
Comment (ColinMaudry) [reply .travis.yml]: Done

File: release-schema.json [hunk 9]
```diff
@@ -26,7 +26,7 @@
           "properties": {
             "url": {
               "title": "Website URL",
-              "description": "The Web address of the website of the organization.",
+              "description": "The Website of the organization.",
```
Comment (jpmckinney) [inline release-schema.json]: Lowercase "website".

File: README.md [hunk 10]
```diff
@@ -1,6 +1,8 @@
 # European Union
 
-Implements schemas and codes that are specific to the European law.
+Implements fields and codes that are specific to the European law.
```
Comment (jpmckinney) [inline README.md]: Remove "the".

File: release-schema.json [hunk 11]
```diff
@@ -48,7 +48,7 @@
       "properties": {
         "valueCalculationMethod": {
           "title": "Value calculation method",
-          "description": "Any details relevant to the value of the concession according to article 8(3) of the directive 2014/23/EU on concession contracts.",
+          "description": "Any details relevant to the value of the concession according to article 8(3) of directive 2014/23/EU on concession contracts.",
```
Comment (jpmckinney) [inline release-schema.json]: Capitalize "Directive".

Comment (ColinMaudry) [reply release-schema.json]: Done

File: README.md (hunk 10 above)
Comment (ColinMaudry) [reply README.md]: Done in README and extension.json `.description`.

File: README.md (hunk 3 above)
Comment (ColinMaudry) [reply README.md]: Done. I've removed BT-10 and BT-610 as they're not related to this extension.

File: release-schema.json (hunk 9 above)
Comment (ColinMaudry) [reply release-schema.json]: Done

**Pull request merged**